simple/src/main/java/org/apache/commons/rdf/simple/experimental/AbstractRDFParser.java

```diff
@@ -240,7 +240,7 @@ public T rdfTermFactory(RDF rdfTermFactory) {
     public T contentType(RDFSyntax rdfSyntax) throws IllegalArgumentException {
         AbstractRDFParser<T> c = clone();
```

Comment:
No, but cloning means the `RDFParser` is immutable and can be reused in a thread-safe manner, e.g. if parsing many Turtle files in the same folder. So what is worse..:

* a) Creating 1000 RDFParser instances, setting all parameters every time, and remember not to reuse/share in multiple threads.
* b) Create a single base RDFParser instance with the common settings, then specialize only what is needed for each file to use in any thread at any time. (e.g. called from a within a `Stream.parallell().map()`

I think the overhead of cloning here is pretty much equivalent, e.g. `1000` instances, or `1005` instances.  

If you are parsing just a single file then I don't think the inefficiency of the cloning makes a big difference - that is unless you are making a very snappy UI or something.

Let's change this to mutable only if benchmark numbers show a problem?